src/coreclr/src/jit/hwintrinsicxarch.cpp

```diff
@@ -1173,19 +1173,209 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
 
 GenTree* Compiler::impSSEIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* sig)
 {
-    GenTree* retNode  = nullptr;
-    GenTree* op1      = nullptr;
-    GenTree* op2      = nullptr;
-    GenTree* op3      = nullptr;
-    GenTree* op4      = nullptr;
-    int      simdSize = HWIntrinsicInfo::lookupSimdSize(this, intrinsic, sig);
+    GenTree*  retNode  = nullptr;
+    GenTree*  op1      = nullptr;
+    GenTree*  op2      = nullptr;
+    int       simdSize = HWIntrinsicInfo::lookupSimdSize(this, intrinsic, sig);
+    var_types baseType = TYP_UNKNOWN;
 
     // The Prefetch and StoreFence intrinsics don't take any SIMD operands
     // and have a simdSize of 0
     assert((simdSize == 16) || (simdSize == 0));
 
     switch (intrinsic)
     {
+        case NI_SSE_CompareGreaterThan:
+        {
+            assert(sig->numArgs == 2);
+            op2      = impSIMDPopStack(TYP_SIMD16);
+            op1      = impSIMDPopStack(TYP_SIMD16);
+            baseType = getBaseTypeOfSIMDType(sig->retTypeSigClass);
+            assert(baseType == TYP_FLOAT);
+
+            if (compOpportunisticallyDependsOn(InstructionSet_AVX))
+            {
+                retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, op2, gtNewIconNode(14), NI_AVX_Compare, baseType,
+                                                   simdSize);
+            }
+            else
+            {
+                retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op2, op1, NI_SSE_CompareLessThan, baseType, simdSize);
```

Comment:
So idea here is to use AVX `vcmpps` if supported, otherwise, flip the arguments and use SSE `cmpps`? This is done because the AVX instruction guaranteed to be more performant even for the same vector size?

Reply:
Yes. `Avx` has direct support for the comparison while `Sse` has to emulate it by doing a different comparison with flipped arguments (and possibly accounting for "copy upper bits")